### ISSUE TITLE
W442: worth the trouble?

### ISSUE THREAD

**Issue by TotalVerb** [tonyhffong/Lint.jl#139]
Currently,

```
W442 &: bit-wise in a boolean context. (&,|) do not have short-circuit behavior
```

is raised when bitwise operators are used in boolean contexts. While it's great that Lint points out how they do not have short-circuit behaviour, I fail to see why this is a problem. If someone uses `&` and `|` in boolean contexts, they are almost certainly doing it intentionally to avoid a branch, resulting in increased performance. I don't know if it is common at all to accidentally use the single `&` or `|` when a short-circuiting operator is desired.

**Comment by Michael-Klassen** [comment]
How about we change it to an `INFO` level warning?

**Comment by Michael-Klassen** [comment]
I was thinking of adding `# noqa` comment support like in flake8, which might also solve these kinds of issues.
